internal/exec/util/file.go

```diff
@@ -33,62 +36,108 @@ const (
 )
 
 type File struct {
-	Path     types.Path
-	Contents []byte
-	Mode     os.FileMode
-	Uid      int
-	Gid      int
+	io.ReadCloser
+	hash.Hash
+	Path types.Path
+	Mode os.FileMode
+	Uid  int
+	Gid  int
+	hash string
```

Comment:
Perhaps name this expectedhash or some thing else more descriptive, otherwise it isn't clear what it is compared to the embedded Hash